### ISSUE TITLE
[Bug]: MotionCam Outdoor 2/4 PhOD photo entities missing

### ISSUE THREAD

**Issue by Cris59264** [bvis/aegis-hass#499]
### Describe the bug

Hello,

I am using Aegis for Ajax with a MotionCam Outdoor 2/4 PhOD.

The device is correctly detected in Home Assistant as:
Motion Cam Outdoor Two Four Phod

Photo on Demand works correctly in the Ajax app, but in Home Assistant/Aegis I do not get:
- a camera entity
- a "take photo" button/entity

On my indoor MotionCam PhOD, these entities are available and work correctly, so the issue seems specific to the Outdoor 2/4 PhOD model.

I have attached the Home Assistant diagnostic file for the device.

Thanks!

### Steps to reproduce

1. Add a MotionCam Outdoor 2/4 PhOD through Aegis for Ajax.
2. Open the device in Home Assistant.
3. Check the available entities.
4. No camera entity or "take photo" button is created.
5. Photo on Demand works correctly from the Ajax app.

[config_entry-aegis_ajax-01KTP3PYAPNSJ2TSMMYBSZX9J2.json](https://github.com/user-attachments/files/32071029/config_entry-aegis_ajax-01KTP3PYAPNSJ2TSMMYBSZX9J2.json)

### Expected behavior

I expect the MotionCam Outdoor 2/4 PhOD to expose the same Photo on Demand entities as my indoor MotionCam PhOD: - a camera entity - a "take photo" button/entity The requested photo should then be viewable directly in Home Assistant. 

### Home Assistant version

2026.9.1

### Integration version

1.19.1

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

No

### Relevant logs

```text
HTS connection error in listen: Connection closed by remote

Device stream kept dropping while idle — reducing gRPC keepalive from 240s to 120s to keep the connection warm

Device stream error for space ..., reconnecting in 5s
```

**Comment by bvis** [comment]
Thanks — the diagnostics settled this in one round, so there is nothing more I need from you.

The cause is a stale list rather than anything about your devices. The integration keeps a list of MotionCam families that receive a camera entity and a capture button, and that list predates the refresh that took the modelled device families from 33 to 107. `motion_cam_outdoor_two_four_phod` was never added to it, so your two Outdoor 2/4 PhOD units were modelled as detectors — motion, tamper, delay — and then skipped for the photo entities. Your indoor unit is a different family that was already on the list, which is exactly why it works. Your dump is what made this a paired comparison rather than a guess: both families sitting on the same hub, same account, same permissions, same version, one getting the entities and the other not.

Fixed in #502, which is green and waiting to merge. I will post here when it is installable as a beta. Then:

1. HACS → Aegis for Ajax → three-dot menu → Redownload, with *Show beta versions* enabled, and pick the new beta.
2. Restart Home Assistant.
3. Open one of the Outdoor 2/4 PhOD devices. You should now see a camera entity and a *Take a photo* button.
4. Press the button once and confirm the photo appears.

Two things worth knowing. A capture spends the detector's own battery, so treat the button as a deliberate action rather than something to poll from an automation. And the camera entity has nothing to show until the first capture or the first camera alarm, so it will look empty right after the upgrade.

The three log lines you pasted are unrelated to this. An idle device stream dropping and reconnecting is expected, and the keepalive line is the integration adjusting to it.

One more thing: your report also answers the open question on #472 for this family. That issue asked for exactly what you gave — a family name plus confirmation that the Ajax app itself offers Photo on Demand for it. The other unmapped MotionCam families are still waiting on the same evidence, so if anyone reading this owns one of them, that thread is the place.

**Comment by bvis** [comment]
@Cris59264 this is now installable as **`1.20.0-beta.3`**.

1. HACS → Aegis for Ajax → three-dot menu → **Redownload**, enable *Show beta versions*, and pick `1.20.0-beta.3`. If it is not listed yet, give HACS a few minutes and refresh.
2. Restart Home Assistant.
3. Open one of the two Outdoor 2/4 PhOD devices. You should now see a camera entity and a *Take a photo* button, the same pair your indoor unit already has.
4. Press the button once and confirm the photo arrives.

That fourth step is the whole confirmation, and unlike most things in this project you can do it on demand rather than waiting for an event. A reply saying it works, or saying what you see instead, is all I need.

I am leaving this issue open until you confirm. Two things that are expected and not faults: the camera entity will look empty until that first capture, since there is nothing stored for the device yet, and a capture draws on the detector's own battery, so it is worth treating the button as a deliberate action rather than something an automation presses on a schedule.

Thanks for the report. The diagnostics file is what made this solvable in one round instead of several.

**Comment by Cris59264** [comment]
Hi,

I installed *1.20.0-beta.3* and restarted Home Assistant.

Good news: the *camera entity and the “Take a photo” button now appear
correctly* on my Outdoor 2/4 PhOD devices. 👍

However, the photo capture itself does not complete. When I press *Take a
photo*, Home Assistant eventually displays the following message:

“The photo capture did not complete. The hub may have rejected the command
or returned no image; check the logs for more details.”

I also checked the *Ajax app*, and no photo or corresponding photo event
appears there. I tried twice with the same result.

For comparison, when I use the same function with my *indoor MotionCam PhOD*,
the photo is successfully taken and appears in the Ajax app.

So, on the Outdoor 2/4 PhOD, the new entities are now exposed correctly,
but the actual Photo on Demand request does not seem to reach or complete
on the Ajax side.

Let me know if you would like me to provide the Aegis logs or another
diagnostics dump from this test.

Thanks!

**Comment by bvis** [comment]
That is a real defect and you found it in under an hour. Fixed in **`1.20.0-beta.4`**, just published.

Half of what beta.3 did was right and half was wrong. The entities are created from one list, and the capture request that actually goes to the hub carries a device type from a **separate** list. That second list is coarse, with only *indoor*, *outdoor* and *Fibra base* as options. I widened the first list and not the second, so the request told the hub your outdoor detector was an indoor MotionCam, and the hub refused it. Your indoor unit is on that second list explicitly, which is exactly why it kept working. The detail that pinned this down was yours: nothing appeared in the Ajax app either, so the hub never took a photo at all, which rules out everything downstream of the request.

1. HACS → Aegis for Ajax → three-dot menu → **Redownload**, *Show beta versions* enabled, pick `1.20.0-beta.4`.
2. Restart Home Assistant.
3. Press *Take a photo* on one of the Outdoor 2/4 PhOD devices.
4. Report what happens.

Good news on what to expect. Your diagnostics show push notifications configured, connected and having delivered 55 messages, so the photo should arrive **in Home Assistant** and not only in the Ajax app. Worth mentioning because the issue form recorded push as not configured, and the dump says otherwise. The dump is right.

If it still fails, the exact error text matters, because three different causes share one message. Sending the log line would settle it in one round. Add this to `configuration.yaml`, restart, press the button once, and look for a line containing *Photo capture failed*:

```yaml
logger:
  logs:
    custom_components.aegis_ajax: debug
```

That line carries the hub's raw answer, which names the actual reason. No need to send a whole log, just that line.

Thanks again. Two rounds from report to fix is only possible because you attached the diagnostics and then told me precisely what you saw, including the part about the Ajax app.

**Comment by Cris59264** [comment]
Hi,

Success! 🎉

I installed *1.20.0-beta.4*, restarted Home Assistant, and tested Photo on
Demand on *both of my MotionCam Outdoor 2/4 PhOD devices*.

It now works correctly on both cameras. The photo is successfully captured
and appears in *both Home Assistant and the Ajax app*.

So the fix in beta.4 has completely resolved the issue on my setup.

Thanks again for the incredibly quick fix! 👍

**Comment by bvis** [comment]
Confirmed on both units, with the photo arriving in Home Assistant and in the Ajax app. That closes it.

Worth recording what made this two rounds instead of five. The diagnostics dump you attached to the report named the exact device family and showed a working indoor MotionCam on the same hub, which is what turned "it does not work" into a comparison with a single variable. Then, when beta.3 half-fixed it, you said the photo was missing **from the Ajax app too** — that one sentence ruled out everything downstream of the request and pointed straight at the device type being sent to the hub.

Closing as fixed. Both fixes are on `main` and ship in the next stable release, `1.20.0`; until that is out, `1.20.0-beta.4` or later is where they live.

One related note: your report also settled one of the eight detector families listed in #472, which tracks the same gap for the MotionCam models nobody has confirmed yet. If you ever add another MotionCam variant, that is the issue to mention it on.
